ui/appui-react/src/appui-react/statusfields/SnapMode.tsx

```diff
@@ -95,9 +111,11 @@ export function SnapModeField(props: SnapModeFieldProps) {
     // eslint-disable-next-line deprecation/deprecation
     (state) => state?.configurableUiState.snapMode
   );
-  const snapMode = props.snapMode ?? reduxSnapMode ?? SnapMode.NearestKeypoint;
 
-  const snapModes: SnapModeFieldEntry[] = [
+  const snapModeChoices = props.snapModes ?? [];
```

Comment:
I'd say if an empty array is provided - no snaps should be rendered.

Reply:
I thought a bit about that and having no snap rendered doesnt really makes sense. SnapModeField is a component to display snap modes. I cannot think of a use case for using it to display nothing. This sounds more like wrong usage of the component to me.
This, with the rule of having at least one snap mode active at all time, make me think that always displaying modes is a must.

Reply:
Well, it's up to the consumer to configure correctly. It will make your implementation simpler and produce the expected results.

Reply:
Unfortunately, it makes the code more complex. 
It raises the case, what's the default snap mode, when no snap modes are available ?
![image](https://github.com/user-attachments/assets/41c7b40c-3594-42d0-9d04-b30812dc081e)

Maybe, `undefined` or `Keypoint`, but both cases add complexity down the line. 
The code is simpler if we can assume that at least one snap mode is available at all time. 
I can still change it if you want, but I have the feeling that we would add complexity for a feature that has no value whatsoever.

Reply:
`SnapModeField` is just a component that renders some menu options. It should not be aware of application constraints, like how many snaps are active, it will still render depending on props passed to it (please ignore the redux store, which complicates this component). Let's render nothing if an empty array is passed in.

Reply:
So, what's the snap mode when an empty array is passed in ? undefined of some default snap mode ?

Reply:
The application determines what is the active snap mode and it should provide correct props to the component i.e. the component might not display "correct" snap mode depending on what props are provided.